aiida/orm/nodes/data/structure.py

```diff
@@ -759,44 +763,53 @@ def __init__(
                 self.set_pymatgen_molecule(pymatgen_molecule)
 
         else:
-            if cell is None:
-                cell = [[1., 0., 0.], [0., 1., 0.], [0., 0., 1.]]
-
             if pbc is None:
+                # If a cell is provided by the user, we default to pbc
+                # If not
                 pbc = [True, True, True]
-
             self.set_cell(cell)
             self.set_pbc(pbc)
 
     def get_dimensionality(self):
         """
         This function checks the dimensionality of the structure and
-        calculates its length/surface/volume
-        :return: returns the dimensionality and length/surface/volume
+        calculates its length/surface/volume.
+
+        If no cell is specified, the length/surface/volume is 0.
+
+        :return: returns a dictionary with keys "dim" (dimensionality integer), "label" (dimensionality label)
+            and "value" (numerical length/surface/volume).
         """
 
         import numpy as np
 
         retdict = {}
 
-        cell = np.array(self.cell)
         pbc = np.array(self.pbc)
         dim = len(pbc[pbc])
 
         retdict['dim'] = dim
         retdict['label'] = self._dimensionality_label[dim]
 
+        if dim not in (0, 1, 2, 3):
+            raise ValueError(f'Dimensionality {dim} must be one of 0, 1, 2, 3')
+
+        if self.cell == _DEFAULT_CELL:
```

Comment:
I think we want here also an `if dim != 0: raise ...`: if we don't have a cell but have some PBC, we have a problem.